src/AstGen.zig

```diff
@@ -5871,6 +5937,7 @@ fn forExpr(
     rl: ResultLoc,
     node: Ast.Node.Index,
     for_full: Ast.full.While,
+    is_statement: bool,
```

Comment:
enum instead of bool?

Reply:
For public functions absolutely, but this is a private function used in only two places so I don't think it matters too much.